File: JavaPools/PoolsLibrary/src/main/java/ru/urfu/javapools/poolslibrary/pool/wrappers/statemonitoring/ObjectLifetimeData.java
package ru.urfu.javapools.poolslibrary.pool.wrappers.statemonitoring;

public class ObjectLifetimeData<TK> {

    private TK _key;
    private long _сreationTimeStamp;
    private long _lastUsageTimeStamp;

    public ObjectLifetimeData(TK key) {
    	
        _key = key;
        _сreationTimeStamp = System.currentTimeMillis();
        _lastUsageTimeStamp = System.currentTimeMillis();
    }

    public void update() {  	
    	_lastUsageTimeStamp = System.currentTimeMillis();
    }

    @Override
    public String toString() {
        return String.format("C='%s' LU='%s'",
        					  _сreationTimeStamp, _lastUsageTimeStamp);
    }

	public TK getKey() {
		return _key;
	}

	public long getCreationTimeStamp() {
		return _сreationTimeStamp;
	}

	public long getLastUsageTimeStamp() {
		return _lastUsageTimeStamp;
	}
	
	public ObjectLifetimeData<TK> setCreationTimeStamp(long creationTimeStamp) {
		_сreationTimeStamp = creationTimeStamp;
		return this;		
	}
	
	public ObjectLifetimeData<TK> setLastUsageTimeStamp(long lastUsageTimeStamp) {
		_lastUsageTimeStamp = lastUsageTimeStamp;
		return this;		
	}
}

File: CSharpPools/PoolsLibrary/Pool/Wrappers/StateMonitoring/ObjectLifetimeData.cs
using System;

namespace PoolsLibrary.Pool.Wrappers.StateMonitoring
{
    class ObjectLifetimeData<TK>
    {
        public TK Key { get; private set; }
        public DateTime CreationTimeStamp { get; internal set; }
        public DateTime LastUsageTimeStamp { get; internal set; }

        private const string _debugDatetimeFormat = "HH:mm:ss:fff";

        public ObjectLifetimeData(TK key)
        {
            Key = key;
            CreationTimeStamp = DateTime.Now;
            LastUsageTimeStamp = DateTime.Now;
        }

        public ObjectLifetimeData<TK> GetUpdated()
        {
            return new ObjectLifetimeData<TK>(Key)
            {
                CreationTimeStamp = this.CreationTimeStamp,
                LastUsageTimeStamp = DateTime.Now,
            };
        }

        public override string ToString()
        {
            return string.Format("C='{0}' LU='{1}'",
                CreationTimeStamp.ToString(_debugDatetimeFormat),
                LastUsageTimeStamp.ToString(_debugDatetimeFormat));
        }
    }
}